dex/src/main/java/io/crate/data/RowsBatchIterator.java

```diff
@@ -37,22 +36,25 @@
     private final Iterable<Row> rows;
     private Iterator<Row> it;
     private Row currentRow = BatchIterator.OFF_ROW;
+    private final InputList rowData;
 
-    public static BatchIterator empty() {
-        return newInstance(Collections.emptyList());
-    }
-
-    public static BatchIterator newInstance(Iterable<Row> rows) {
-        return new CloseAssertingBatchIterator(new RowsBatchIterator(rows));
+    public static BatchIterator newInstance(Iterable<Row> rows, int numCols) {
+        return new CloseAssertingBatchIterator(new RowsBatchIterator(rows, numCols));
```

Comment:
would be probably however it is complicated since the row instance of the iterable might not always stay the same, so it would be just an overhead. i think we should leave this here to make it explicit that this is used only for cases where the row objects differ